h/services/email.py

```diff
@@ -18,22 +19,35 @@
 DAILY_SENDER_MENTION_LIMIT = 100
 
 
-@dataclass(frozen=True)
+@dataclass
 class EmailData:
     recipients: list[str]
     subject: str
     body: str
     tag: EmailTag
     html: str | None = None
     subaccount: str | None = None
+    reply_to: str | None = None
+    from_name: str | None = None
 
-    @property
-    def message(self) -> pyramid_mailer.message.Message:
+    def message(self, default_sender: str) -> pyramid_mailer.message.Message:
         extra_headers: dict[str, str] = {"X-MC-Tags": self.tag}
+
         if self.subaccount:
             extra_headers["X-MC-Subaccount"] = self.subaccount
+
+        if self.reply_to:
+            extra_headers["Reply-To"] = self.reply_to
+
         subject = " ".join(self.subject.splitlines())
+
+        if self.from_name:
+            sender = formataddr((self.from_name, parseaddr(default_sender)[1]))
```

Comment:
We don't allow groups to customise the `From` _email address_, only the name part. The group's `email_from_name` replaces only the name part from the default sender. So for example if the group's `email_from_name` is `"Group From Name"` and the `MAIL_DEFAULT_SENDER` envvar is `"Default Name <default_email_address@example.com>"` then `sender` will end up as `"Group From Name <default_email_address@example.com>"`.